Userland/Libraries/LibGfx/PNGWriter.cpp

```diff
@@ -163,7 +163,9 @@ void PNGWriter::add_IDAT_chunk(Gfx::Bitmap const& bitmap)
     ByteBuffer uncompressed_block_data;
     uncompressed_block_data.ensure_capacity(bitmap.size_in_bytes() + bitmap.height());
 
-    Pixel const dummy_scanline[bitmap.width()] {};
+    u8 dummy_buffer[bitmap.width() * sizeof(Pixel)];
+    memset(dummy_buffer, 0, sizeof(dummy_buffer));
+    auto dummy_scanline = reinterpret_cast<Pixel const*>(dummy_buffer);
```

Comment:
There's no reason to use a raw byte array, let the compiler work it out for you:
```suggestion
    Pixel dummy_scanline[bitmap.width()];
    memset(dummy_scanline, 0, sizeof(dummy_scanline));
```

Reply:
gcc doesn’t like that:
```
/home/krkk/src/serenity/Userland/Libraries/LibGfx/PNGWriter.cpp: In member function 'void Gfx::PNGWriter::add_IDAT_chunk(const Gfx::Bitmap&, Compress::ZlibCompressionLevel)':
/home/krkk/src/serenity/Userland/Libraries/LibGfx/PNGWriter.cpp:167:11: error: 'void* memset(void*, int, size_t)' clearing an object of non-trivial type 'union Gfx::Pixel'; use assignment or value-initialization instead [-Werror=class-memaccess]
  167 |     memset(dummy_scanline, 0, sizeof(dummy_scanline));
      |     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krkk/src/serenity/Userland/Libraries/LibGfx/PNGWriter.cpp:140:23: note: 'union Gfx::Pixel' declared here
  140 | union [[gnu::packed]] Pixel {
      |                       ^~~~~
```

Although, I could change it once again, to use `ARGB32` instead of `u8` and remove the `sizeof(Pixel)`, but the cast has to stay anyway. So I don’t think if there’s any advantage of doing this:
```cc
ARGB32 dummy_buffer[bitmap.width()];
memset(dummy_buffer, 0, sizeof(dummy_buffer));
auto dummy_scanline = reinterpret_cast<Pixel const*>(dummy_buffer);
```

Reply:
You're right, but I also just realised there's no point to any of this anyways, `Pixel`'s members are initialised, the compiler will zero out this array for you.

Reply:
Note the inserted call to memset: https://godbolt.org/

Reply:
> You're right, but I also just realised there's no point to any of this anyways, `Pixel`'s members are initialised, the compiler will zero out this array for you.

I’m not so sure. It started producing glitchart.

![screenshot-2022-07-10-17-37-30](https://user-images.githubusercontent.com/16520278/178155830-39537fe3-94cf-4eeb-a22b-b0e12511f949.png)